wemake_python_styleguide/visitors/tokenize/statements.py

```diff
@@ -256,3 +259,194 @@ def _post_visit(self) -> None:
+            self._potential_violation or
+            self.seen_clause_in_line
+        )
+        return self._check_violation(token)
+
+    def handle_if(self, token: tokenize.TokenInfo) -> bool:
+        """
+        Handles 'if' tokens inside brackets.
+
+        In order to rule out this if statement being part of a ternary operator
+        in the action statement of a comprehension, we ensure that we've already
+        passed the action statement, as once a for statement appears any
+        following if statements will be the conditional statement of the
+        comprehension.
+        """
+        if self._seen_for:
+            self._seen_if_in_line = True
+
+            self._potential_violation = (
+                self._potential_violation or
+                self.seen_clause_in_line
+            )
+            return self._check_violation(token)
+        return True
+
+    def handle_in(self, token: tokenize.TokenInfo) -> bool:
+        """
+        Adds a violation when a for...in statement is split across lines.
+
+        Because of the overloaded nature of 'in', we have to rule
+        out several edge cases before we can say that this is a violation:
+        First, we only want to consider this split for...in statement
+        in the case that we've seen a for statement but not on this line.
+        Second, we don't want to catch an 'in' in the conditional clause of the
+        comprehension, so we make sure we haven't seen an if statement on
+        this line.
+        """
+        if self._seen_for:
+            if not self._seen_for_in_line:
+                if not self._seen_if_in_line:
+                    self._reported = True
+                    return False
+        return True
+
+    def _check_violation(self, token: tokenize.TokenInfo) -> bool:
+        """Checks if current environment state implies a violation."""
+        if self._seen_nl:
+            if self._potential_violation and not self._reported:
+                self._reported = True
+                return False
+        return True
+
+
+@final
+@alias('visit_any_left_bracket', (
+    'visit_lsqb',
+    'visit_lbrace',
+    'visit_lpar',
+))
+@alias('visit_any_right_bracket', (
+    'visit_rsqb',
+    'visit_rbrace',
+    'visit_rpar',
+))
+class InconsistentComprehensionVisitor(BaseTokenVisitor):
+    """
+    Visitor for checking inconsistent comprehension syntax.
+
+    Checks if comprehensions either use only one line or inserts a newline
+    for each clause (i.e. bracket, action, for loop(s), and conditional)
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        """
+        Initializes stack of bracket contexts.
+
+        Creates an empty stack for bracket contexts to accomodate for nested
+        comprehensions.
+        """
+        super().__init__(*args, **kwargs)
+        self._bracket_stack: List[InconsistentComprehensionContext] = []
+        self._ctxt: Optional[InconsistentComprehensionContext] = None
```

Comment:
NIP: I have never seen the usage of `ctxt` as a `context` abbreviation, what do you say about renaming it to `_current_context` or `_current_ctx`?

Reply:
Agreed, makes the variable's purpose more understandable. Changed to `_current_ctx`